resolver/resolver.cc

```diff
@@ -2989,8 +2989,15 @@ class ResolveTypeMembersAndFieldsWalk {
             // make sure that it's even possible to be valid.
             auto *cnst = ast::cast_tree<ast::ConstantLit>(cast->typeExpr);
             ENFORCE(cnst != nullptr, "Rewriter should always use const for typeExpr, which should now be resolved");
-            if (!cnst->symbol.isClassOrModule() || cnst->symbol.asClassOrModuleRef().data(ctx)->flags.isModule ||
-                cnst->symbol.asClassOrModuleRef().data(ctx)->typeArity(ctx) > 0) {
+
+            // Dealias to mimic how type_syntax parsing will work for constant lit node (e.g., want
+            // to allow class aliases).
+            auto dealiased = cnst->symbol.dealias(ctx);
+
+            // Don't want the dealias to apply to the type alias, because MyTypeAlias.new is a runtime error.
+            if (cnst->symbol.isTypeAlias(ctx) || !dealiased.isClassOrModule() ||
```

Comment:
I assume we have a test for the `TypeAlias.new` case somewhere, ideally?  If not, can we add one?

Reply:
I'm going to play it cool and say it was a test for my reviewer to see if they're paying attention 🙈 

will add